File: windows/app/Core/WorkFilter.cs
using System;
using System.Collections.Generic;
using System.Linq;

namespace QPVPN.Core;

/// <summary>
/// Второй переключатель: рабочие ресурсы.
///
/// Адреса заложены в приложение. Включён — они идут через VPN, выключен —
/// напрямую. Списки те же, что в версии для Android.
/// </summary>
public static class WorkFilter
{
    public sealed record Resource(string Title, string Url)
    {
        /// <summary>Узел из ссылки: без схемы, пути и порта.</summary>
        public string Host
        {
            get
            {
                var text = Url;
                var scheme = text.IndexOf("://", StringComparison.Ordinal);
                if (scheme >= 0) text = text[(scheme + 3)..];
                var slash = text.IndexOf('/');
                if (slash >= 0) text = text[..slash];
                var colon = text.IndexOf(':');
                if (colon >= 0) text = text[..colon];
                return text;
            }
        }
    }

    public static readonly IReadOnlyList<Resource> Resources = new List<Resource>
    {
        new("Ka", "https://135.106.142.73/Ka"),
        new("Ka_old", "https://135.106.142.73/Ka_old"),
    };

    public static IReadOnlyList<string> Hosts =>
        Resources.Select(resource => resource.Host).Distinct().ToList();

    public static int Count => Resources.Count;
}

File: android/app/src/main/java/kz/qpvpn/model/WorkFilter.kt
package kz.qpvpn.model

/**
 * Второй переключатель: рабочие ресурсы.
 *
 * Адреса заложены в приложение. Включён — они идут через VPN, выключен —
 * напрямую, с домашнего адреса. Переключатель работает независимо от главного
 * фильтра и сильнее общих правил: даже когда вся российская зона идёт мимо
 * туннеля, включённый переключатель заворачивает эти адреса в туннель.
 *
 * Маршрутизация в Android идёт по адресам, а не по ссылкам: обе ссылки ведут
 * на один узел, поэтому в маршруты уходит один адрес.
 */
object WorkFilter {

    data class Resource(val title: String, val url: String) {
        /** Узел из ссылки: без схемы, пути и порта. */
        val host: String
            get() = url.substringAfter("://").substringBefore('/').substringBefore(':')
    }

    val resources: List<Resource> = listOf(
        Resource("Ka", "https://135.106.142.73/Ka"),
        Resource("Ka_old", "https://135.106.142.73/Ka_old"),
    )

    /** Узлы без повторов — в таком виде они уходят в маршрутизацию. */
    val hosts: List<String> = resources.map { it.host }.distinct()

    val count: Int get() = resources.size
}
